### DISCUSSION TITLE
DomiRank Centrality

### DISCUSSION THREAD

**Discussion by visunf** [networkx/networkx#7321] [Ideas]
The algorithm module of networkx has APIs for calculating degrees, eigenvectors, closeness, and betweenness centrality. Could you please add an API for calculating DomiRank Centrality for the convenience of users? DomiRank Centrality was proposed in an article "DomiRank Centrality reveals structural fragility of complex networks via node dominance" published in Nature Communications by Marcus Engsig et al. in 2024. I think integrating this API in networkx will be beneficial to both users and the promotion of networkx. There is source code provided by the author here, and I hope you can consider integrating DomiRank Centrality into networkx.

ref. :
Engsig, M., Tejedor, A., Moreno, Y. et al. DomiRank Centrality reveals structural fragility of complex networks via node dominance. Nat Commun 15, 56 (2024). https://doi.org/10.1038/s41467-023-44257-0

https://github.com/mengsig/DomiRank

**Comment by mengsig** [comment]
Hello Visunf,

I am sorry I did not see this post before. I can start writing an API for integrating DomiRank to networkx and see start direct communication with them to see if they would be willing to accept my implementation. In the worst case, it will probably accelerate their development. 

Best,
Marcus

**Comment by visunf** [comment]
Thank you for your contribution to networkx!

mengsig ***@***.***> 于2024年4月30日周二 16:13写道：

**Reply by mengsig** [reply]
Hello Visunf,

I have made the git pull now in Networkx. You can follow status updates here:

https://github.com/networkx/networkx/pull/7443

Best,
Marcus
